cmd/simulator/worker/worker.go

```
@@ -0,0 +1,316 @@
+	transferAmount   = big.NewInt(1)
+	workDelay        = time.Duration(100 * time.Millisecond)
+	retryDelay       = time.Duration(500 * time.Millisecond)
+
+	chainID     *big.Int
+	signer      types.Signer
+	feeCap      *big.Int
+	priorityFee *big.Int
+
+	maxTransferCost  *big.Int
+	requestAmount    *big.Int
+	minFunderBalance *big.Int
+)
+
+func setupVars(cID *big.Int, bFee uint64, pFee uint64) {
+	chainID = cID
+	signer = types.LatestSignerForChainID(chainID)
+	priorityFee = new(big.Int).SetUint64(pFee * params.GWei)
+	feeCap = new(big.Int).Add(new(big.Int).SetUint64(bFee*params.GWei), priorityFee)
+
+	maxTransferCost = new(big.Int).Mul(new(big.Int).SetUint64(transferGasLimit), feeCap)
+	maxTransferCost = new(big.Int).Add(maxTransferCost, transferAmount)
+
+	requestAmount = new(big.Int).Mul(maxTransferCost, big.NewInt(100))
+	minFunderBalance = new(big.Int).Add(maxTransferCost, requestAmount)
+}
+
+func createWorkers(ctx context.Context, keys []*key.Key, endpoints []string, desiredWorkers int) (*worker, []*worker, error) {
+	var master *worker
+	var workers []*worker
+	for _, k := range keys {
+		worker, err := newWorker(k, endpoints[rand.Intn(len(endpoints))])
+		if err != nil {
+			return nil, nil, err
+		}
+		if err := worker.fetchBalance(ctx); err != nil {
+			return nil, nil, err
+		}
+		if err := worker.fetchNonce(ctx); err != nil {
+			return nil, nil, fmt.Errorf("could not get nonce: %w", err)
+		}
+		log.Printf("loaded worker %s (balance=%s nonce=%d)\n", worker.k.Address.Hex(), worker.balance.String(), worker.nonce)
+
+		switch {
+		case master == nil:
+			master = worker
+		case new(big.Int).Sub(master.balance, worker.balance).Sign() < 0:
+			workers = append(workers, master)
+			master = worker
+		default:
+			workers = append(workers, worker)
+		}
+	}
+
+	var err error
+	if master == nil {
+		master, err = newWorker(nil, endpoints[rand.Intn(len(endpoints))])
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable to create master: %w", err)
+		}
+	}
```

Should we return an error here or will this require that the user of the load simulator sends funds to the generated key? If the latter, can we add a comment here explaining why this isn't a fatal error.

---

there is a separate loop to wait for funds